base/irrationals.jl

```diff
@@ -3,24 +3,32 @@
 ## general machinery for irrational mathematical constants
 
 """
-    Irrational <: Real
+    AbstractIrrational <: Real
 
-Irrational number type.
+Number type representing an exact irrational value.
 """
-struct Irrational{sym} <: Real end
+abstract type AbstractIrrational <: Real end
```

Comment:
Could be added to the "Abstract number types" section of `doc/src/stdlib/numbers.md`

Reply:
Okay, done.